From df810d5b9cf8c486a069434ecc339731a4b89f4b Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Mon, 13 Sep 2021 10:38:32 +0200
Subject: [PATCH] Respect CMAKE_ARGS if set by the environment

---
 setup.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 7b1c4677..44a8cd15 100644
--- a/setup.py
+++ b/setup.py
@@ -94,7 +94,10 @@ class CMakeBuild(setuptools.command.build_ext.build_ext):
         # Set Python_EXECUTABLE instead if you use PYBIND11_FINDPYTHON
         # EXAMPLE_VERSION_INFO shows you how to pass a value into the C++ code
         # from Python.
-        cmake_args = [
+        cmake_args = []
+        if "CMAKE_ARGS" in os.environ:
+            cmake_args += [item for item in os.environ["CMAKE_ARGS"].split(" ") if item]
+        cmake_args += [
             "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={}".format(extdir),
             "-DPYTHON_EXECUTABLE={}".format(sys.executable),
             "-DEXAMPLE_VERSION_INFO={}".format(self.distribution.get_version()),
-- 
2.32.0

